project.js

```diff
@@ -273,3 +281,159 @@ function guid() {
+
+    $.get(baseUrl + "rest/project/list", function () {}).done(function (data) {
+        projCount = parseInt(data.total);
+        var projPages = Math.floor(projCount / 100) + 1
+
+        for (var pc = 0; pc < projPages; pc++) {
+            $.get(baseUrl + "rest/project/list?offset=" + (pc * 100).toString(10), function () {}).done(function (data) {
```

Comment:
Question: Line 347 obtains a complete list of project details (sans code). Why is this loop going back out to get the project details again, albeit in smaller groupings? We should do one or the other, but not both.

Reply:
Line 347 does not actually return a full list.  It does return a total count, which we need to know, but the endpoint only returns details for up to 100 projects (maximum) at a time.  Any users that have more than 100 projects will require multiple requests to the REST endpoint., which is why we have to loop back with an offset of 100 * n.

Reply:
There are only 40 user accounts out of the approximately 18,000 that have in excess of 100 projects. Would it make more sense to loop on the $.get() in line 352, pulling in up to 100 projects at a time. If the call results in 100 projects, increment the offset by 100 and loop to get the next group? In 99.8% of the requests, the code will never actually loop. The code in its current state will request the first 100 projects twice over the vast majority of the user accounts.